qiskit/transpiler/__init__.py

```diff
@@ -19,8 +19,364 @@
+     `optimization_level=2` and `optimization_level=3`).
+
+
+   The choice of initial layout is extremely important when:
+
+   1. Computing the number of SWAP operations needed to map the input circuit onto the device
+      topology.
+
+   2. Taking into account the noise properties of the device.
+
+
+   The choice of `initial_layout` can mean the difference between getting a result,
+   and getting nothing but noise.
+
+   Lets see what layouts are automatically picked at various optimization levels.  The modified
+   circuits returned by :func:`qiskit.compiler.transpile` have this initial layout information
+   in them, and we can view this layout selection graphically using
+   :func:`qiskit.visualization.plot_circuit_layout`:
+
+   .. jupyter-execute::
+
+      from qiskit import QuantumCircuit, transpile
+      from qiskit.visualization import plot_circuit_layout
+      from qiskit.test.mock import FakeVigo
+      backend = FakeVigo()
+
+      ghz = QuantumCircuit(3, 3)
+      ghz.h(0)
+      ghz.cx(0,range(1,3))
+      ghz.barrier()
+      ghz.measure(range(3), range(3))
+      ghz.draw(output='mpl')
+
+
+   - **Layout Using Optimization Level 0**
+
+      .. jupyter-execute::
+
+         new_circ_lv0 = transpile(ghz, backend=backend, optimization_level=0)
+         plot_circuit_layout(new_circ_lv0, backend)
+
+
+   - **Layout Using Optimization Level 3**
+
+      .. jupyter-execute::
+
+         new_circ_lv3 = transpile(ghz, backend=backend, optimization_level=3)
+         plot_circuit_layout(new_circ_lv3, backend)
+
+
+   It is completely possible to specify your own initial layout.  To do so we can
+   pass a list of integers to :func:`qiskit.compiler.transpile` via the `initial_layout`
+   keyword argument, where the index labels the virtual qubit in the circuit and the
+   corresponding value is the label for the physical qubit to map onto:
+
+   .. jupyter-execute::
+
+      # Virtual -> physical
+      #    0    ->    3
+      #    1    ->    4
+      #    2    ->    2
+
+      my_ghz = transpile(ghz, backend, initial_layout=[3, 4, 2])
+      plot_circuit_layout(my_ghz, backend)
+
+   .. raw:: html
+
+      <br>
+
+
+.. container:: toggle
+
+   .. container:: header
+
+      **Mapping Circuits to Hardware Topology**
+
+   In order to implement a CNOT gate between qubits in a quantum circuit that are not directly
+   connected on a quantum device one or more SWAP gates must be inserted into the circuit to
+   move the qubit states around until they are adjacent on the device gate map.  Each SWAP
+   gate is decomposed into three CNOT gates on the IBM Quantum devices, and represents an
+   expensive and noisy operation to perform.  Thus, finding the minimum number of SWAP gates
+   needed to map a circuit onto a given device, is an important step (if not the most important)
+   in the whole execution process.
+
+   However, as with many important things in life, finding the optimal SWAP mapping is hard.
```

Comment:
Heh, this is a bit philosophical for documentation :)

Reply:
Well it is supplementary material so all bets are off.